### PR TITLE
Server now parses config file for port number

### PR DESCRIPTION
**Pull request by jasonsliu**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 TEST_SOURCE = config_parser_test.cc config_parser.cc $(GTEST_DIR)/src/gtest_main.cc
 
-all: parser
+all: parser server
 
 parser:	$(PARSER_SOURCE)
-	$(CC) $(CFLAGS) -o $(TARGET) $(PARSER_SOURCE)
+	$(CC) $(CFLAGS) -g -o $(TARGET) $(PARSER_SOURCE)
+
+server: $(SERVER_SOURCE)
```
Comment (ryanvoong) [inline Makefile]: Is the executable supposed to be called `webserver`? See the [assignment 2 spec](https://docs.google.com/document/d/1UG9n7rnITq3XwIc2ijGXXJk0Y2OTwvU6KXz7mTs-yOM/pub)

File: main.cc [hunk 2]
```diff
@@ -2,11 +2,16 @@
...
+  if(argc != 2) {
+    std::cerr << "Usage: " << argv[0] << " config_path" << std::endl;
+    return 1;
+  }
+
   // Creates server object to accept incoming client connections
   try {
     boost::asio::io_service io_service;
-    Server server(io_service, 8118);
+    Server server(io_service, argv[1]);
```
Comment (crystalhsieh) [inline main.cc:14]: If we don't have two arguments, which port number does it run?

Also - we should add in the README about the commands to run :)

File: server.cc [hunk 3]
```diff
@@ -1,17 +1,37 @@
...
+Server::Server(boost::asio::io_service& io_service, const char* filename)
+    : acceptor_(io_service) {
+  
+  NginxConfig config;
+  NginxConfigParser parser;
+  if(!parser.Parse(filename, &config)) {
+    throw "Error parsing config file.";
+  }
+
+  port_ = std::stoi(config.statements_[0]->tokens_[1]);
```
Comment (crystalhsieh) [inline server.cc:21]: please add a comment that we get the port number from the config file :)

**Review by crystalhsieh — COMMENTED** [review]
overall looks good - just add a comment and instructions on how to run

File: main.cc (hunk 2 above)
Comment (jasonsliu) [reply main.cc:14]: 2 arguments are required! Else the server wont run.

File: connection.cc [hunk 4]
```diff
@@ -40,4 +40,6 @@ void Connection::send(std::size_t length) {
           receive();
```
Comment (ryanvoong) [inline connection.cc:40]: should the socket be closed here instead (inside the callback)?

**Review by ryanvoong — COMMENTED** [review]
looks good

File: main.cc (hunk 2 above)
Comment (crystalhsieh) [reply main.cc:14]: alrighty sounds good :+1:

File: connection.cc (hunk 4 above)
Comment (crystalhsieh) [reply connection.cc:40]: would that mean we would close the socket as soon as we get a response? 
Connection::send() will hang until the callback receives something, right?

**Review by crystalhsieh — APPROVED** [review]
looks good to me - after ryan's stuff is resolved

**Pull request merged**